### ISSUE TITLE
Improve the generated class

### ISSUE THREAD

**Issue by nirodg** [nirodg/easyrs#7]
Due each method is an independent test each method we should persist first the data and afterwords proceed to execute the proper crud operation depending on the method which is being tested.

An example about how **getAll()** shoud look like
```java
  @Test
  @InSequence(1)
  public void getAll() {
  	
  	UserDto persistDto = Data.importData();
  	persistDto = client.put(dto);
  	
  	Assert.assertNotNull(persistDto);
  	
  	List<UserDto> getPersistedEntities = client.get();
  	Assert.assertNotNull(getPersistedEntities);
  	
  	Assert.assertEquals(1, getPersistedEntities.size());
  	
  }
```

**Comment by nirodg** [comment]
Generator improved (from now one will be called ClassBuilder unless a better name is given) so it's able so far to generate the followed content for an annotated interface:

```java
package com.app.test;

import com.dbrage.lib.easyrs.arquillian.Container;
import com.dbrage.lib.easyrs.arquillian.utils.Checker;
import com.dbrage.lib.easyrs.client.Client;
import java.util.ArrayList;
import java.util.List;
import org.jboss.arquillian.container.test.api.Deployment;
import org.jboss.arquillian.junit.Arquillian;
import org.jboss.arquillian.junit.InSequence;
import org.jboss.shrinkwrap.api.spec.WebArchive;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import static com.dbrage.lib.easyrs.processor.enums.ClientRequest.ALL;
import static com.dbrage.lib.easyrs.processor.enums.ClientRequest.DELETE;
import static com.dbrage.lib.easyrs.processor.enums.ClientRequest.POST;
import static com.dbrage.lib.easyrs.processor.enums.ClientRequest.PUT;

@RunWith(Arquillian.class)
@SuppressWarnings("unchecked")
public class UserRestTestEndpoint extends Container<UserDto> {

  private Client client;

  @Before
  public void setUp() {
    client = new Client();
  }

  @Deployment
  public static final WebArchive deployment() {
    WebArchive deployment = getDeployment(UserRest.class);
    return deployment;
  }

  @Test
  @InSequence(1)
  public void getAll() {
    // TODO: Map an entity given the json file
    // TODO: Persist the mocked entity to have access to the GUID/ID

    List<com.app.test.UserDto> entities = (ArrayList<com.app.test.UserDto>) getData(ALL);
    List<com.app.test.UserDto> fetchedEntities = (ArrayList<com.app.test.UserDto>) client.getAll();
    Assert.assertEquals(entities.size(), fetchedEntities.size());
  }

  @Test
  @InSequence(2)
  public void create() {
    // TODO: Map an entity given the json file
    // TODO: Persist the mocked entity to have access to the GUID/ID

    com.app.test.UserDto entity = (com.app.test.UserDto) getData(PUT);
    Assert.assertNotNull(entity);
    com.app.test.UserDto fetchedEntity = (com.app.test.UserDto) client.put(entity);
    Assert.assertNotNull(fetchedEntity);
    Assert.assertEquals(entity, fetchedEntity);
  }

  @Test
  @InSequence(3)
  public void update() {
    // TODO: Map an entity given the json file
    // TODO: Persist the mocked entity to have access to the GUID/ID

    com.app.test.UserDto entity = (com.app.test.UserDto) getData(POST);
    Assert.assertNotNull(entity);
    com.app.test.UserDto fetchedEntity = (com.app.test.UserDto) client.post(entity.getGuid(), entity);
    Assert.assertNotNull(fetchedEntity);
    Assert.assertEquals(entity, fetchedEntity);
  }

  @Test
  @InSequence(4)
  public void delete() {
    // TODO: Map an entity given the json file
    // TODO: Persist the mocked entity to have access to the GUID/ID

    com.app.test.UserDto entity = (com.app.test.UserDto) getData(DELETE);
    Assert.assertNotNull(entity);
    com.app.test.UserDto fetchedEntity = (com.app.test.UserDto) client.delete(entity.getGuid(), entity);
    Assert.assertNull(fetchedEntity);
  }

}
```

Next steps are to polish minor details and complete the TODO comments.

**Comment by nirodg** [comment]
Committed fix for minor bug. 
Currently the methods are not created in order, must check that.

**Comment by nirodg** [comment]
With commit  c1a8fd7 the methods are ordered properly

**Comment by nirodg** [comment]
With commit  c353b82762417c8c1913e5abc51474fed73980b9 the entity is persisted before is checked with the expected result

**Comment by nirodg** [comment]
Issue closed with merge #18 into develop's branch

**Comment by nirodg** [comment]
Merge #18  included additional class files, fixed with merge  683e9e1
